src/c2ffi/integration/cl-ppcre.lisp

```diff
@@ -0,0 +1,36 @@
+;;;; -*- Mode: lisp; indent-tabs-mode: nil -*-
+;;;
+;;; cl-ppcre.lisp --- Adds cl-ppcre support to name matching.
+;;;
+;;; Copyright (C) 2015, Attila Lendvai <attila@lendvai.name>
+;;;
+;;; Permission is hereby granted, free of charge, to any person
+;;; obtaining a copy of this software and associated documentation
+;;; files (the "Software"), to deal in the Software without
+;;; restriction, including without limitation the rights to use, copy,
+;;; modify, merge, publish, distribute, sublicense, and/or sell copies
+;;; of the Software, and to permit persons to whom the Software is
+;;; furnished to do so, subject to the following conditions:
+;;;
+;;; The above copyright notice and this permission notice shall be
+;;; included in all copies or substantial portions of the Software.
+;;;
+;;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+;;; EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+;;; MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+;;; NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+;;; HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+;;; WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+;;; DEALINGS IN THE SOFTWARE.
+;;;
+
+(in-package #:cffi/c2ffi)
+
+(defun make-rule-matcher/cl-ppcre (pattern)
+  (let ((scanner (cl-ppcre:create-scanner pattern)))
+    (named-lambda cffi/c2ffi/cl-ppcre-rule-matcher
+        (string)
+      (funcall scanner string 0 (length string)))))
+
+(setf *make-matcher-from-rule-hook* 'make-rule-matcher/cl-ppcre)
```

Comment:
yep, but i was hoping that cl-ppcre is backwards compatible enough with a string search, but thinking again it's not. especially that the default rule matcher is an ends-with...

i'm not happy with the ppcre dependency, but i don't have a sane idea for an alternative either... will sleep on it and see if something comes up.

re checking in the generated bindings: this is actually not a bad idea, because the .spec files practically define the .lisp files (modulo cffi/c2ffi generator changes, or its parameters in the c2ffi-file in the asdf system).

trouble is that the user libs need to straight out :defsystem-depends-on cffi/c2ffi to be able to define the system itself, so a hard dependency on cl-ppcre would come with cffi/c2ffi.

maybe user libs could have two systems:
 - one with a full cffi/c2ffi dependency, and the ability to regenerate the binding
 - and another one that merely loads the binding, assuming that it has been checked into the repo.

i'll sleep one on this, too. i think it's actually a good idea to keep the generated lisp files also next to the .spec files, and let the users decide whether they check it into their repos or not.